Makefile.codegen

```diff
@@ -15,15 +20,21 @@ test-codegen: ## Test the code geneator
 
 # Generate go code using generate directives in files and kubernetes code generation
 # Anything generated by this target should be checked in
-generate: generate-clean build-codegen generate-mocks generate-openapi generate-client ## Generate the Go code (crds, mocks, openapi, client)
+generate: generate-clean build-codegen install-generate-deps generate-mocks generate-openapi generate-client ## Generate the Go code (crds, mocks, openapi, client)
 	@$(MAKE) fmt
 	@echo "Generation complete"
 
-PEGO_MOCK := $(GOPATH)/bin/pegomock
-$(PEGO_MOCK):
-	$(GO) get github.com/petergtz/pegomock/...@v2.2.0
+install-generate-deps:
```

Comment:
I mainly added this for convenieve in conjunction with `tidy-deps` which would otherwise require to run the full code generation which does install the tooling as well in case it is missing.